### PR TITLE
Fixes component base README.md KEP link

### LINKED ISSUE #76851: Component Base KEP Link in readme.
**What happened**:
Looks like KEP organization changed and the component-base kep in the readme doesn't link anymore.

**What you expected to happen**:
Link to go to the right document.

**How to reproduce it (as minimally and precisely as possible)**:
click link - https://github.com/kubernetes/component-base/blob/master/README.md

**Anything else we need to know?**:
n/a

**Environment**:
n/a

### PR DESCRIPTION
**Pull request by christopherhein**
Signed-off-by: Christopher Hein <me@chrishein.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Fixes documentation link.

**Which issue(s) this PR fixes**:
Fixes #76851

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/component-base/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Purpose
 
-Implement KEP 32: https://github.com/kubernetes/enhancements/blob/master/keps/sig-cluster-lifecycle/0032-create-a-k8s-io-component-repo.md
+Implement KEP 32: https://git.k8s.io/enhancements/keps/sig-cluster-lifecycle/wgs/0032-create-a-k8s-io-component-repo.md
```
Comment (stewart-yu) [inline staging/src/k8s.io/component-base/README.md]: @christopherhein  Thanks for your fix
would yu like change the link to `https://github.com/kubernetes/enhancements/blob/master/keps/sig-cluster-lifecycle/wgs/0032-create-a-k8s-io-component-repo.md` ?

Comment (christopherhein) [reply staging/src/k8s.io/component-base/README.md]: Prefer that over the short link? All ends up the same place

Comment (christopherhein) [reply staging/src/k8s.io/component-base/README.md]: @stewart-yu done. 👍

**Review by luxas — APPROVED** [review]
/approve
Thanks @christopherhein!

**Pull request merged**